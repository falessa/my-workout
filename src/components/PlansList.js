import React from 'react';
import { StyleSheet, View, Text, FlatList, Button } from 'react-native';
import PlanCard from './PlanCard/PlanCard';
import PlanCardsSeparator from './PlanCardsSeparator'
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'stretch',
        justifyContent: 'center',
        width: '85%',
    }
})

const PlansList = ({ plans, navigation }) => {
    const { t } = useTranslation()

    return (
        <View style={styles.container}>
            <Text>{t('plansTitle')}</Text>
            <FlatList
                data={plans}
                keyExtractor={x => String(x.id)}
                renderItem={({ item }) => <PlanCard planData={item} onPress={() => navigation.navigate('PlanDetailScreen', { planId: item.id } )}/> }
                ItemSeparatorComponent={PlanCardsSeparator}
            />
            <Button title={t('addPlan')} onPress={() => navigation.navigate('AddPlanName')}/>
        </View>
    )
}

export default PlansList;