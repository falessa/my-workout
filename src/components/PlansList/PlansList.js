import React from 'react';
import { StyleSheet, View, Text, FlatList, Button, TouchableOpacity} from 'react-native';
import PlanCard from '../PlanCard/PlanCard';
import PlanCardsSeparator from '../PlanCardsSeparator/PlanCardsSeparator'
import MainButton from '../MainButton'
import { useTranslation } from 'react-i18next';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'stretch',
        justifyContent: 'center',
        width: '85%',
    }
})

const PlansList = ({ plans, navigation }) => {
    const { t } = useTranslation()

    return (
        <View style={styles.container}>
            <Text>{t('plansTitle')}</Text>
            <FlatList
                data={plans}
                keyExtractor={x => String(x.id)}
                renderItem={({ item }) => <PlanCard planData={item} onPress={() => navigation.navigate('PlanDetailScreen', { planId: item.id } )}/> }
                ItemSeparatorComponent={PlanCardsSeparator}
            />
            <MainButton onPress={() => navigation.navigate('AddPlanName')} text={t('addPlan')}/>
        </View>
    )
}

export default PlansList;