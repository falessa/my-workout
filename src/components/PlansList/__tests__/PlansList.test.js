import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import PlansList from '../PlansList';

describe('PlansList tests', () => {
    const props = {
        plans: [
            {
                id: "92fx027h-0f27-38g2-93s1-b93829l29u38",
                name: "Workout test plan 1",
                daysPerWeek: "2",
                creationDate: "October 19th 2020"
            },
            {
                id: "03ge983g-2a92-01q7-83r4-a79032h87s10",
                name: "Workout test plan 2",
                daysPerWeek: "4",
                creationDate: "October 21st 2020"
            }
        ],
        navigation: () =>{
            // nothing
        }
    }

    const wrapper = create(<PlansList {...props} />)

    test('Render PlansList', () => {
        expect(wrapper).toMatchSnapshot()
    })
})