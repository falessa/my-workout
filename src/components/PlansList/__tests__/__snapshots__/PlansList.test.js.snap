// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlansList tests Render PlansList 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "width": "85%",
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "creationDate": "October 19th 2020",
          "daysPerWeek": "2",
          "id": "92fx027h-0f27-38g2-93s1-b93829l29u38",
          "name": "Workout test plan 1",
        },
        Object {
          "creationDate": "October 21st 2020",
          "daysPerWeek": "4",
          "id": "03ge983g-2a92-01q7-83r4-a79032h87s10",
          "name": "Workout test plan 2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 10,
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 20,
              "padding": 10,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Workout test plan 1
            </Text>
            <Text>
              2
               
              daysWeek
            </Text>
            <Text>
              createdAt
               
              October 19th 2020
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 15,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 10,
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 20,
              "padding": 10,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Workout test plan 2
            </Text>
            <Text>
              4
               
              daysWeek
            </Text>
            <Text>
              createdAt
               
              October 21st 2020
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#386fa4",
        "borderRadius": 20,
        "margin": 15,
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      addPlan
    </Text>
  </View>
</View>
`;
